templates/CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+
+Each PR should pass the tests and the linter to be accepted.
+
+```bash
+# Tests
+$ <Add Docker command to run MeiliSearch here>
+$ <Add the right command here>
+# Linter
+$ <Add the right command here>
+# Linter with fixing
+$ <Add the right command here>
+```
+
+### Release Process
+
+MeiliSearch tools follow the [Semantic Versioning Convention](https://semver.org/).
+
+#### Automated Changelogs
+
+For each PR merged on `master`, a GitHub Action is running and updates the next release description as a draft release in the [GitHub interface](https://github.com/meilisearch/XXX/releases). If you don't have the permission to this repository, you will not be able to see the draft release until the release will be published.
+
+The draft release description is therefore generated and corresponds to all the PRs titles since the previous release. This means each PR should only do one change and the title should be desriptive of this change.
+
+About this automation:
+- As the draft release description is generated on every push on `master`, don't change it manually until the final release publishment.
+- If you don't want a PR to appear in the release changelogs: set the label `skip-changelog`. We used to remove PRs updating the README or the CI (except for big changes).
+- If the changes you are doing in the PR are breaking: set the label `breaking-change`. In the release tag, the minor will be increased instead of the patch. The major will never be changed until [MeiliSearch](https://github.com/meilisearch/MeiliSearch) is stable.
+- If you did any mistake, for example the PR is already closed but you forgot to add a label or you misnamed your PR, don't panic: change what you want in the closed PR and run the job again.
+
+*More information about [release drafter](https://github.com/release-drafter/release-drafter), used to automate these steps.*
+
+#### How to Publish the Release
+
+You must do a PR modifying the file `[<Add file name here>]`(\<Add relative path to the file here\>) with the right version.
+
+```<Add the language here>
+<Add the line to change, e.g.: "version": X.X.X>
+```
+
+Once the changes are merged on `master`, you can publish the current draft release via the [GitHub interface](https://github.com/meilisearch/XXX/releases).
+
+A GitHub Action will be triggered and push the package on [\<Add the platform name here, e.g. npm>](\<Add URL to the platform here\>).
+
+## Git Guidelines
+
+### Git Branches
+
+All changes must be made in a branch and submitted as PR.
+We do not enforce any branch naming style, but please use something descriptive of your changes.
+
+### Git Commits
+
+As minimal requirements, your commit message should be:
+- to be capitalized
+- not to finish by a dot or any other punctuation character (!,?)
+- to start with a verb so that we can read your commit message this way: "This commmit will ..." where "..." is the commit message.
+  e.g.: "Fix the home page button" or "Add more tests for create_index method"
```

Comment:
```suggestion
As minimal requirements, your commit message should:
- be capitalized
- not finish by a dot or any other punctuation character (!,?)
- start with a verb so that we can read your commit message this way: "This commmit will ..." where "..." is the commit message.
  e.g.: "Fix the home page button" or "Add more tests for create_index method"
```